src/app/components/game-components/rectangular-game-component/RectangularGameComponent.ts

```diff
@@ -18,17 +19,32 @@ export abstract class RectangularGameComponent<R extends Rules<M, S, L>,
 
     public board: Table<P>;
 
+    public getWidth(): number {
+        return this.board[0].length;
+    }
+
+    public getHeight(): number {
+        return this.board.length;
+    }
+
     public getViewBox(): string {
-        return (-0.5 * this.STROKE_WIDTH) + ' ' +
-            (-0.5 * this.STROKE_WIDTH) + ' ' +
-            (this.board.length * this.SPACE_SIZE + this.STROKE_WIDTH) + ' ' +
-            (this.board[0].length * this.SPACE_SIZE + this.STROKE_WIDTH);
+        const left: number = -0.5 * this.STROKE_WIDTH;
+        const up: number = -0.5 * this.STROKE_WIDTH;
+        const right: number = this.getWidth() * this.SPACE_SIZE + this.STROKE_WIDTH;
+        const down: number = this.getHeight() * this.SPACE_SIZE + this.STROKE_WIDTH;
+        return ViewBox.fromLimits(left, right, up, down).toSVGString();
     }
-    public getViewBoxWithExtraSpace(bonusLeft: number, bonusRight: number,
-                                    bonusUp: number, bonusDown: number): string {
-        return (-0.5 * this.STROKE_WIDTH - bonusLeft) + ' ' +
-            (-0.5 * this.STROKE_WIDTH - bonusUp) + ' ' +
-            (this.board.length * this.SPACE_SIZE + this.STROKE_WIDTH + bonusRight) + ' ' +
-            (this.board[0].length * this.SPACE_SIZE + this.STROKE_WIDTH + bonusDown);
+
+    public getViewBoxWithExtraSpace(bonusLeft: number,
+                                    bonusRight: number,
+                                    bonusUp: number,
+                                    bonusDown: number)
+    : string
+    {
+        const left: number = -0.5 * this.STROKE_WIDTH - bonusLeft;
+        const up: number = -0.5 * this.STROKE_WIDTH - bonusUp;
+        const right: number = (this.getWidth() * this.SPACE_SIZE + this.STROKE_WIDTH + bonusRight);
+        const down: number = (this.getHeight() * this.SPACE_SIZE + this.STROKE_WIDTH + bonusDown);
```

Comment:
avoir standardThingie and bonussedThingie ?

Reply:
code disparu